src/basic/sysctl-util.c

```diff
@@ -58,6 +58,30 @@ int sysctl_write(const char *property, const char *value) {
 
         log_debug("Setting '%s' to '%s'", p, value);
 
+        if (shadow) {
+                _cleanup_free_ char *k = NULL, *v = NULL, *cur_k = NULL, *cur_v = NULL;
+                int r;
+
+                k = strdup(p);
+                if (!k)
+                        return -ENOMEM;
+
+                v = strdup(value);
+                if (!v)
+                        return -ENOMEM;
+
+                cur_k = hashmap_get2(*shadow, k, (void**)&cur_v);
```

Comment:
i am very confused now. hashmap_get2() returns the *value* as return value, and the *key* in the 3rd arg. Your variable names suggest the opposite.

Reply:
Well, I was assuming that `hashmap_replace()` keeps the old key, but actually it replaces it.
I'll do it even differently,